docs/care/CEP/Draft/0012-careplan.md

```diff
@@ -0,0 +1,247 @@
+# CEP-12 Care Plan
+
+### References
+
+- [https://ohcnetwork.notion.site/Care-Plan-12ae6e667ece80e8b1abfdc82b3450f2](https://ohcnetwork.notion.site/Care-Plan-12ae6e667ece80e8b1abfdc82b3450f2)
+- [https://www.hl7.org/fhir/careplan.html](https://www.hl7.org/fhir/careplan.html)
+
+## Wireframes
+
+- [https://www.figma.com/design/wE4aB7XELuASk2dKBUKToZ/Care-Plan-Wireframe?node-id=0-1&t=Av3ebeUiCTmRoAan-1](https://www.figma.com/design/wE4aB7XELuASk2dKBUKToZ/Care-Plan-Wireframe?node-id=0-1&t=Av3ebeUiCTmRoAan-1)
+
+## Implementation Overview
+
+> A Care Plan is a coordinated map of healthcare activities designed to address specific patient health concerns. At its core, it's an organized collection of intended activities that healthcare providers plan to perform, are performing, or have performed for a patient.
+
+To visualize better, lets create a django model for care plan
+
+```py
+class CarePlan(BaseModel):
+    status = Int(choices) # draft|active|completed|on-hold|cancelled|entered-in-error|unknown
+    intent = Int(choices) # proposal|plan|order|option|directive
+    title = CharField()
+    description = TextField()
+    start_date = DateField()
+    end_date = DateField()
+    subject/patient = FK(Patient) # wont use if we use encounter
+    encounter = FK(Consultation) # not sure if consultations will stay.
+    custodian/faciliy = FK(Facility) # wont use if we use encounter
+    addresses = FK(ICD11Diagnoses) #or SNOMED going forward
+    notes = TextField()
+```
+
+SNOMED-CT is not yet implemented with care, so we would default to using ICD 11 diagnosis until SNOMED is configured.
+
+The care plan will have activities associated with it. These will be done to accomplish the goals in a care plan.
+
+```py
+class CarePlanActivity(BaseModel): # or just Activity if it is generic
+    performed = BooleanField()
+    activity_task = FK(Task)
```

Comment:
Where is `Task` defined 🤔

Reply:
based on presumtption of Task being predefined. Should I include Task and ServiceRequest in the scope?